invenio_rdm_migrator/load/postgresql/table_generators/single.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Migrator is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Single table generator module."""
+
+from ..transactions import Operation
+from .base import CSVTableGenerator, MemoryTableGenerator
+
+
+class SingleTableGenerator(CSVTableGenerator):
+    """Contains only one table to yield."""
+
+    def __init__(self, table, pks=None, post_load_hooks=None):
+        """Constructor."""
+        assert not isinstance(table, (list, dict))
+
+        super().__init__(tables=[table], pks=pks, post_load_hooks=post_load_hooks)
+
+    def _generate_rows(self, data, **kwargs):
+        """Yield generated rows."""
+        table = self.tables[0]
+        yield table(**data)
+
+
+# FIXME: not a fan of duplicating this amount of code...
+# _generate_rows could be composable on a table generator
+# decoupling output (mem/csv) and row generation
+class MemorySingleTableGenerator(MemoryTableGenerator):
```

Comment:
It is needed, but the implementation and interface is completely different than the one of CSV... another broken api signature